cyral/resource_cyral_repository_user_account.go

```diff
@@ -0,0 +1,596 @@
+				HttpMethod: http.MethodDelete,
+				CreateURL: func(d *schema.ResourceData, c *client.Client) string {
+					return fmt.Sprintf(
+						"https://%s/v1/repos/%s/userAccounts/%s",
+						c.ControlPlane,
+						d.Get("repository_id").(string),
+						d.Id(),
+					)
+				},
+			},
+		),
+		Importer: &schema.ResourceImporter{
+			StateContext: func(
+				ctx context.Context,
+				d *schema.ResourceData,
+				i interface{},
+			) ([]*schema.ResourceData, error) {
+				ids, err := unmarshalComposedID(d.Id(), "/", 2)
+				if err != nil {
+					return nil, fmt.Errorf(
+						"failed to unmarshal ID: %v",
+						err,
+					)
+				}
+				repositoryID := ids[0]
+				userAccountID := ids[1]
+				err = d.Set("repository_id", repositoryID)
+				if err != nil {
+					return nil, fmt.Errorf(
+						"failed to set 'repository_id': %v",
+						err,
+					)
+				}
+				d.SetId(userAccountID)
+				return []*schema.ResourceData{d}, nil
+			},
+		},
+		Schema: map[string]*schema.Schema{
+			"id": {
+				Description: "ID of this resource in Cyral environment.",
+				Type:        schema.TypeString,
+				Computed:    true,
+			},
+
+			"repository_id": {
+				Description: "ID of the repository.",
+				Required:    true,
+				Type:        schema.TypeString,
+				ForceNew:    true,
+			},
+
+			"name": {
+				Description: "The name of the User Account.",
+				Required:    true,
+				Type:        schema.TypeString,
+			},
+
+			"auth_database_name": {
+				Description: "For User Accounts associated with a MongoDB cyral_repository, " +
+					"This value is only meaningful for User Accounts associated with " +
+					"MongoDB.",
+				Optional: true,
+				Type:     schema.TypeString,
+			},
+
+			"approval_config": {
+				Description: "Configurations related to Approvals.",
+				Optional:    true,
+				Type:        schema.TypeSet,
+				MaxItems:    1,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"automatic_grant": {
+							Description: "If true, it is possible to create an approval with an automatic grant" +
+								"associated with this User Account.",
+							Optional: true,
+							Type:     schema.TypeBool,
+						},
+						"max_auto_grant_duration": {
+							Description: "The maximum duration in seconds for the lifespan of the approval, " +
+								"if automatic_grant is true. E.g.: '2000s', '3000.5s'",
+							Optional: true,
+							Type:     schema.TypeString,
+						},
+					},
+				},
+			},
+
+			"auth_scheme": {
+				Description: "Credential option. List of supported types: " +
+					supportedTypesMarkdown(allAuthSchemes),
+				Required: true,
+				Type:     schema.TypeSet,
+				MaxItems: 1,
+				MinItems: 1,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"environment_variable": {
+							Description: "Credential option to set the repository user account from " +
+								"Environment Variable.",
+							Optional: true,
+							Type:     schema.TypeSet,
+							MaxItems: 1,
+							MinItems: 1,
```

Comment:
@VictorGFM I don't think it is possible to use `ExactlyOneOf` in a schema nested inside TypeSet. I have struggled with this in the past and found no way around it. See [this comment](https://github.com/hashicorp/terraform-plugin-sdk/blob/main/helper/schema/schema.go#L278-L283)

Reply:
If we use TypeList with `MaxItems: 1` for `auth_scheme`, then it is possible. I think this would make sense. There is no difference between TypeList with `MaxItems: 1` and TypeSet with `MaxItems: 1` from the user's perspective, so it should be OK. LMK your thoughts.

Reply:
Yes, I was also running into issues with `ExactlyOneOf` in a TypeSet. Even though TypeList is generally supposed to be used for ordered lists, as @Yowgf mentioned, this doesn't matter if there can only be one element. I will try that out.